channels/rdpecam/client/encoding.c

```diff
@@ -205,22 +216,21 @@ static BOOL ecam_encoder_compress_h264(CameraDeviceStream* stream, const BYTE* s
 		}
 	}
 
-	/* get buffers for YUV420P */
-	if (h264_get_yuv_buffer(stream->h264, srcLineSizes[0], size.width, size.height, yuv420pData,
-	                        yuv420pStride) < 0)
-		return FALSE;
-
-	/* convert from source format to YUV420P */
+	/* convert from source format to YUV420P or NV12 */
 	if (!ecam_init_sws_context(stream, pixFormat))
 		return FALSE;
 
 	const BYTE* cSrcSlice[4] = { srcSlice[0], srcSlice[1], srcSlice[2], srcSlice[3] };
-	if (sws_scale(stream->sws, cSrcSlice, srcLineSizes, 0, (int)size.height, yuv420pData,
-	              (int*)yuv420pStride) <= 0)
+	if (sws_scale(stream->sws, cSrcSlice, srcLineSizes, 0, (int)size.height, stream->swsOutData,
```

Comment:
@oleg0421 hmm, wouldn´t moving this part into the encoder function solve your issue to have different encoding formats?
this way you could just feed native format and then convert to whatever you need internally?

Reply:
you could even expose a `out_data_free` function to wrap the cleanup calls and not link `ffmpeg` directly this way